pylint/checkers/exceptions.py

```diff
@@ -131,7 +131,7 @@ def _is_raising(body: list) -> bool:
         "try-except-raise block!",
     ),
     "W0707": (
-        "Consider explicitly re-raising using the 'from' keyword",
+        "Consider explicitly re-raising using 'raise NewError(...) from old_error'",
```

Comment:
It would probably be a lot more user-friendly if we passed `old_error` as an argument and showed the actual error that was caught. I think we have that information when we `add_message`, right?

Reply:
```diff
diff --git a/pylint/checkers/exceptions.py b/pylint/checkers/exceptions.py
index 7e182c3dc..26024f8aa 100644
--- a/pylint/checkers/exceptions.py
+++ b/pylint/checkers/exceptions.py
@@ -336,6 +336,12 @@ class ExceptionsChecker(checkers.BaseChecker):
         if containing_except_node.name is None:
             # The `except` doesn't have an `as exception:` part, meaning there's no way that
             # the `raise` is raising the same exception.
+            if isinstance(containing_except_node.type, nodes.Name):
+                name_of_old_error = containing_except_node.type.name
+            elif isinstance(containing_except_node.type, nodes.Tuple):
+                name_of_old_error = ", ".join(
+                    n.name for n in containing_except_node.type.elts
+                )
             self.add_message("raise-missing-from", node=node)
         elif isinstance(node.exc, nodes.Call) and isinstance(node.exc.func, nodes.Name):
             # We have a `raise SomeException(whatever)`.
```

? We have access to the wrapping except handler and can determine the node name from it.

There are three other places at which the message is raised but I think we should be able to get the names there as well.

Reply:
Say we have this:
```python
try:
    1 / 0
except ZeroDivisionError:
    # +1: [raise-missing-from]
    raise KeyError
```
``name_of_old_error`` would be ``ZeroDivisionError`` when we would want it to be ``e`` or something so the end result is this:
```python
try:
    1 / 0
except ZeroDivisionError as e:
    raise KeyError from e
```
So the message should be something like `` "Consider explicitly re-raising using 'raise KeyError from old_error'",`` and if we have ``except ZeroDivisionError as exc:`` in the original code it should be  ``Consider explicitly re-raising using 'raise KeyError from exc"``, right ?

Reply:
Ah. I think we should potentially rephrase the message. What about:
``Consider explicitly re-raising using 'except %s as exc' and 'raise from exc'``

That should work for both situations right?

Reply:
👍 I don't know if we can do ``Consider explicitly re-raising using 'except %s as exc' and 'raise%s from exc'``  with the second args being ``KeyError`` or ``ValueError("my message"), but ``Consider explicitly re-raising using 'except %s as exc' and 'raise ... from exc'`` is probably good enough ?

Reply:
Yeah I think the `...` works. Messages shouldn't be too long I think so that seems like a nice middle ground.

Is `exc` the most common name to do this? Docs seems to use `err` and `exc`: https://docs.python.org/3/tutorial/errors.html

Reply:
A small refactor permitted to have a nice little message. At this point we might as well autofix files 😄 Something to consider when we have less than 10 issues opened :trollface: